larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -754,19 +828,36 @@ void LArMCParticleHelper::SelectGoodCaloHits(const CaloHitList *const pSelectedC
 //------------------------------------------------------------------------------------------------------------------------------------------
 
 void LArMCParticleHelper::SelectParticlesMatchingCriteria(const MCParticleVector &inputMCParticles, std::function<bool(const MCParticle *const)> fCriteria,
-    MCParticleVector &selectedParticles)
+    MCParticleVector &selectedParticles, const bool foldBackHierarchy, bool isTestBeam)
 {
     for (const MCParticle *const pMCParticle : inputMCParticles)
     {
-        if (fCriteria(pMCParticle))
-            selectedParticles.push_back(pMCParticle);
+        if (foldBackHierarchy)
+        {
+            if (!fCriteria(pMCParticle))
+                continue;
+        }
+        else
+        {
+            if(isTestBeam)
```

Comment:
Typically have space after if, e.g. `if (isTestBeam)`

Reply:
sorry! still getting used to the Pandora style